src/java/org/apache/cassandra/index/sai/disk/v3/CassandraDiskAnn.java

```diff
@@ -186,7 +188,9 @@ public CloseableIterator<ScoredRowId> search(VectorFloat<?> queryVector,
             var rr = view.rerankerFor(queryVector, sf);
             ssp = new SearchScoreProvider(asf, rr);
         }
-        var result = searcher.search(ssp, topK, threshold, ordinalsMap.ignoringDeleted(acceptBits));
+        var result = searcher.search(ssp, limit, rerankK, threshold, context.getAnnRerankFloor(), ordinalsMap.ignoringDeleted(acceptBits));
```

Comment:
How does this impact resumed queries?

Reply:
It's ignored.  Basically if our assumptions were wrong (we didn't find enough good results the first time) then our assumption around rerankFloor was not correct and should be thrown out.  

More detail from the jvector internals:

    // Since Astra / Cassandra's usage drives the design decisions here, it's worth being explicit
    // about how that works and why.
    //
    // Astra breaks logical indexes up across multiple physical OnDiskGraphIndex pieces, one per sstable.
    // Each of these pieces is searched independently, and the results are combined.  To avoid doing
    // more work than necessary, Astra assumes that each physical ODGI will contribute responses
    // to the final result in proportion to its size, and only asks for that many results in the initial
    // search.  If this assumption is incorrect, or if the rows found turn out to be deleted or overwritten
    // by later requests (which will be in a different sstable), Astra wants a lightweight way to resume
    // the search where it was left off to get more results.
    //
    // Because Astra uses a nonlinear overquerying strategy (i.e. rerankK will be larger in proportion to
    // topK for small values of topK than for large), it's especially important to avoid reranking more
    // results than necessary.  Thus, Astra will look at the worstApproximateInTopK value from the first
    // ODGI, and use that as the rerankFloor for the next.  Thus, rerankFloor helps avoid believed-to-be-
    // unnecessary work in the initial search, but if the caller needs to resume() then that belief was
    // incorrect and is discarded, and there is no reason to pass a rerankFloor parameter to resume().
    //
    // Finally: in the majority of cases, the initial search() does suffice.  So while we could add the
    // complexity of caching exact scores from candidates that were ultimately evicted from the results,
    // we expect that to be useful only in a small minority of cases -- particularly since we are using

Reply:
Thanks for sharing, makes sense